### PR TITLE
[Refactoring] Better hook memoization

### PR DESCRIPTION
**Pull request by Philipp91**
Context: https://github.com/mui-org/material-ui-pickers/pull/1095#pullrequestreview-247398372

Since a mistake in this code would introduce subtle bugs, depending on how good the test coverage is for these fils, I suggest an extra-thorough review. The changes in each file are technically independent from the other files.

### CODE REVIEW

File: lib/src/_shared/hooks/usePickerState.ts [hunk 1]
```diff
@@ -45,13 +45,10 @@ export function usePickerState(props: BasePickerProps, options: StateHookOptions
   const acceptDate = useCallback(
     (acceptedDate: MaterialUiPickersDate) => {
       setIsOpen(false);
-      props.onChange(acceptedDate);
-
-      if (props.onAccept) {
-        props.onAccept(acceptedDate);
-      }
+      onChange(acceptedDate);
+      onAccept && onAccept(acceptedDate);
```
Comment (dmtrKovalenko) [inline lib/src/_shared/hooks/usePickerState.ts]: please leave if here

File: lib/src/_shared/hooks/usePickerState.ts [hunk 2]
```diff
@@ -74,32 +71,33 @@ export function usePickerState(props: BasePickerProps, options: StateHookOptions
...
+  }, [onError, validationError, value]);
 
+  const inputValue = getDisplayDate(date, format, utils, value === null, props);
   const inputProps = useMemo(
     () => ({
       validationError,
-      onClick: () => !props.disabled && setIsOpen(true),
-      inputValue: getDisplayDate(date, format, utils, props.value === null, props),
+      onClick: () => !disabled && setIsOpen(true),
+      inputValue,
```
Comment (dmtrKovalenko) [inline lib/src/_shared/hooks/usePickerState.ts]: here make inputValue as a first key in the object

**Comment by dmtrKovalenko** [comment]
I will clone the branch and will explicitly check if all deps are working properly. But right now LGTM. Thanks for help 👍

**Comment by Philipp91** [comment]
Should I address the review comments now, or are you doing that yourself on the cloned branch?

**Pull request merged**